### PR TITLE
Fixes for Qt6

### PR DESCRIPTION
**Pull request by Herrie82**
Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

### CODE REVIEW

**Comment by Herrie82** [comment]
This has been tested to build OK with Qt 5.15 and 6.3. 

I don't have an older Qt 5.x to test.

**Comment by Thaodan** [comment]
> I don't have an older Qt 5.x to test.

Please read Qt docs or the headers of your Qt version they state since when something is deprecated.

File: libconnman-qt/vpnmodel.h [hunk 1]
```diff
@@ -34,9 +34,9 @@
 #define VPNMODEL_H
 
 #include <QAbstractListModel>
+#include "vpnmanager.h";
 
 class VpnModelPrivate;
-class VpnManager;
```
Comment (Thaodan) [inline libconnman-qt/vpnmodel.h]: Don't remove the partial class that this intentional.

File: libconnman-qt/vpnmanager.h [hunk 2]
```diff
@@ -34,10 +34,10 @@
 #define VPNMANAGER_H
 
 #include <QObject>
+#include "vpnconnection.h"
 
 class VpnManagerPrivate;
 class VpnManager;
-class VpnConnection;
```
Comment (Thaodan) [inline libconnman-qt/vpnmanager.h]: Don't remove the partial class that this intentional.

**Comment by Thaodan** [comment]
Please change each port away from a deprecated into an individual commit.
The changes in themselves only have indirectly something to with Qt 6.
Also the commit style doesn't match the existing commits.

File: libconnman-qt/vpnmodel.h (hunk 1 above)
Comment (Herrie82) [reply libconnman-qt/vpnmodel.h]: Well this needs to be done for Qt6 by the looks of it due to static assertion. See https://stackoverflow.com/questions/66581395/q-property-must-be-fully-defined-error-in-qt-6 for more details.

**Comment by Herrie82** [comment]
> Please change each port away from a deprecated into an individual commit. The changes in themselves only have indirectly something to with Qt 6. Also the commit style doesn't match the existing commits.

Will rework it into separate and clear commits

**Pull request closed without merging**